### PR TITLE
Creating docs for React Training

### LINKED ISSUE #302: React Native Training
**Is your feature request related to a problem? Please describe.**
We currently have very little in the form of classes and/or documentation on React Native even though we are using it in quite a few projects. I would be nice to reuse what we know works.

**Describe the solution you'd like**
I would like us to add more recommended classes, resources, and references to our React documentation. 

Also I would like to ask everyone, what do they think of splitting out the React and React native docs into their own separate headers (or pages)?

### PR DESCRIPTION
**Pull request by mxrcochxvez**
## Changes
1. Creating Docs for React Training.

## Purpose
The purpose of this PR is to add more information about using React/React Native and referencing important features of the Library.

## Learning
For this PR I had to research the exact definitions of each method and keyword that the library uses as well as determining the most useful changes between ES5 and ES6 to include in the document as well as making sure that encouraging Functional components was just as important as showing how similar problems would be solved with Class Based Components.

Here are some documents I referenced
- [https://medium.com/edonec/state-in-react-an-overview-a182675cee2c](https://medium.com/edonec/state-in-react-an-overview-a182675cee2c)
- [https://github.com/uberVU/react-guide/blob/master/props-vs-state.md](https://github.com/uberVU/react-guide/blob/master/props-vs-state.md)
- [https://reactjs.org/docs/getting-started.html](https://reactjs.org/docs/getting-started.html)
- [https://www.w3schools.com/tags/tag_button.asp](https://www.w3schools.com/tags/tag_button.asp)
- [https://www.better.dev/react-prerequisites](https://www.better.dev/react-prerequisites)

Closes #302

### CODE REVIEW

**Review by michaelachrisco — CHANGES_REQUESTED** [review]
@mxrcochxvez Thanks for adding to our processes!

Always assign pr-team to your PRs, per our [code review](https://github.com/Shift3/standards-and-practices/blob/main/standards/code-reviews.md#process) process. Assign your PRs to @Shift3/pr-team in addition to the main devs on this project. This will automatically grab three developers from the PR team and assign them to perform a review. This will help to get a quicker turnaround on your reviews, and will also help this project to get reviews and feedback from team members with different perspectives and skill levels.

**Comment by mxrcochxvez** [comment]
> @mxrcochxvez Thanks for adding to our processes!
> 
> Always assign pr-team to your PRs, per our [code review](https://github.com/Shift3/standards-and-practices/blob/main/standards/code-reviews.md#process) process. Assign your PRs to @Shift3/pr-team in addition to the main devs on this project. This will automatically grab three developers from the PR team and assign them to perform a review. This will help to get a quicker turnaround on your reviews, and will also help this project to get reviews and feedback from team members with different perspectives and skill levels.

Thank you Michael! I went ahead and did this.

File: best-practices/native/react-native/startup/native-code.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # React Native Code Start up Guide
 
 1. [Go here](https://facebook.github.io/react-native/docs/getting-started.html) and make sure you have selected the `Building Projects with Native Code` tab. Follow the CLI installation instructions for your OS.
-2. Take advantage of Bitwise's [Frontend Masters](https://frontendmasters.com/courses/react-native/) course on React Native. This will get you familiar with the basics of React Native's design patters, state management, and common best practices.
-3. Install [Prop Types](https://github.com/facebook/prop-types) library to ensure type checking on your custom component props.
+2. Take advantage of Bitwise's [Frontend Masters](https://frontendmasters.com/courses/react-native/) course on React Native. This will get you familiar with the basics of React Native's design patterns, state management, and best practices.
```
Comment (michaelachrisco) [inline best-practices/native/react-native/startup/native-code.md:4]: Nice spelling/grammar pass here.

File: best-practices/native/react-native/training/README.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+- React is a UI layer for developers to build their applications.
+- Utilizes a virtual DOM to render components.
+- Very small learning curve if you can use ES6.
+
+## React Native
+
+- Uses JSX elements that can *include* HTML elements but generally uses specific native components.
+- React Native is a UI layer **and** a complete framework for building native applications on iOS and Android.
+- Utilizes the device API to render components.
+- Slightly more complex learning curve if you understand ES6, little-to-no learning curve coming from React.
```
Comment (michaelachrisco) [inline best-practices/native/react-native/training/README.md:27]: Can you add more info for developers that want a deep dive here? For example: adding links to our react and react native resources:

https://github.com/Shift3/standards-and-practices/blob/dcd03c29e48a9f0824ab01ab38fe0057a79338ac/best-practices/client-side/framework/react/README.md

https://github.com/Shift3/standards-and-practices/blob/54a24028df8c34e723f5e8399cbcd31002113232/best-practices/native/react-native/README.md

File: best-practices/native/react-native/training/README.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
+# Table of contents
```
Comment (michaelachrisco) [inline best-practices/native/react-native/training/README.md:1]: I believe this page should be added to a parent page such as: https://github.com/Shift3/standards-and-practices/blob/main/best-practices/client-side/framework/react/README or similar.

**Review by michaelachrisco — APPROVED** [review]
Awesome job! Excellent Spelling and Grammar checks here! 

We definitely need more info on react vs react native. Im wondering if we may want to add the training README in some sort of parent README such as a link within https://github.com/Shift3/standards-and-practices/blob/main/best-practices/client-side/framework/react/README.md or something similar. To me this isn't necessarily a blocker.

**Pull request merged**